packages/alchemy/src/react/hooks/useSmartAccountClient.ts

```diff
@@ -0,0 +1,97 @@
+"use client";
+
+import {
+  accountLoupeActions,
+  lightAccountClientActions,
+  multiOwnerPluginActions,
+  pluginManagerActions,
+  type AccountLoupeActions,
+  type LightAccount,
+  type LightAccountClientActions,
+  type MultiOwnerModularAccount,
+  type MultiOwnerPluginActions,
+  type PluginManagerActions,
+} from "@alchemy/aa-accounts";
+import type { Chain, Transport } from "viem";
+import { createAlchemySmartAccountClientFromRpcClient } from "../../client/internal/smartAccountClientFromRpc.js";
+import type {
+  AlchemySmartAccountClient,
+  AlchemySmartAccountClientConfig,
+  BaseAlchemyActions,
+} from "../../client/smartAccountClient";
+import type { SupportedAccounts } from "../../config";
+import type { AlchemySigner } from "../../signer";
+import { useBundlerClient } from "./useBundlerClient.js";
+
+export type UseSmartAccountClientProps<
+  TTransport extends Transport = Transport,
+  TChain extends Chain | undefined = Chain | undefined,
+  TAccount extends SupportedAccounts = SupportedAccounts
+> = Omit<
+  AlchemySmartAccountClientConfig<TTransport, TChain, TAccount>,
+  "rpcUrl" | "chain" | "apiKey" | "jwt" | "account"
+> & { account: TAccount };
```

Comment:
that does mean that the result of `useSmartAccountClient` can return a nullable client, or maybe I can have it return a client that accepts either type of account. the tricky part is adding the actions to the generic client... if no account should that return a client which has both LA and MA actions attached?

Reply:
I went with returning a nullable account and its usage is here:
https://github.com/alchemyplatform/embedded-accounts-demo/pull/4/files#diff-36b09641ed3ec22dbde5def2ae2d515f826edc07f78919d2cc16024701f8a785R33-R36

I kinda like this because it means you can use the `useAccount` hook in the same component as the `useSmartAccountClient` hook. It also means that the developer can await the result of `useAccount` before using the client.

maybe we should bring some of the functionality of `useAccount` into `useSmartAccountClient`. Instead of calling it with an account instance you call it with an account type like with `useAccount` and associated params. That would turn this hook into one line for usage rather than two lines since you always need to `useAccount` to `useSmartAccountClient`

Reply:
ok I ended up doing the last bit here instead. This hook now takes in the same params as `useAccount` + configs for the `smartAccountClient`. devex reflected here: https://github.com/alchemyplatform/embedded-accounts-demo/pull/4/files#diff-36b09641ed3ec22dbde5def2ae2d515f826edc07f78919d2cc16024701f8a785R32-R34

feels much better, but lemme know if you have any reservations with this approach

Reply:
Yeah, I like this! This looks pretty similar to the hook I ended up with in Worth of Words which returns a nullable account object and it works pretty smoothly there.